src/ffi.mjs

```diff
@@ -36,6 +36,17 @@ export function to_fetch_request(request) {
   return new globalThis.Request(url, options);
 }
 
+export function bitarray_request_to_fetch_request(request) {
+  let url = uri_to_string(to_uri(request));
+  let method = method_to_string(request.method).toUpperCase();
+  let options = {
+    headers: make_headers(request.headers),
+    method,
+  };
+  if (method !== "GET" && method !== "HEAD") options.body = request.body.buffer;
+  return new globalThis.Request(url, options);
+}
```

Comment:
All the code here except `.buffer` is the same as the other function, let's extract that into one function.

Reply:
This looks better now